cpp/benchmarks/streaming/ndsh/q17.cpp

```diff
@@ -0,0 +1,1195 @@
+                auto avg_yearly_val = total_sum / 7.0;
+                auto avg_yearly_scalar = cudf::make_numeric_scalar(
+                    cudf::data_type(cudf::type_id::FLOAT64),
+                    chunk_stream,
+                    ctx->br()->device_mr()
+                );
+                static_cast<cudf::numeric_scalar<double>*>(avg_yearly_scalar.get())
+                    ->set_value(avg_yearly_val, chunk_stream);
+
+                std::vector<std::unique_ptr<cudf::column>> result_cols;
+                result_cols.push_back(
+                    cudf::make_column_from_scalar(
+                        *avg_yearly_scalar, 1, chunk_stream, ctx->br()->device_mr()
+                    )
+                );
+
+                co_await ch_out->send(
+                    rapidsmpf::streaming::to_message(
+                        0,
+                        std::make_unique<rapidsmpf::streaming::TableChunk>(
+                            std::make_unique<cudf::table>(std::move(result_cols)),
+                            chunk_stream
+                        )
+                    )
+                );
+            } else {
+                // No data after filtering - send empty result with 0.0
+                auto zero_scalar = cudf::make_numeric_scalar(
+                    cudf::data_type(cudf::type_id::FLOAT64),
+                    chunk_stream,
+                    ctx->br()->device_mr()
+                );
+                static_cast<cudf::numeric_scalar<double>*>(zero_scalar.get())
+                    ->set_value(0.0, chunk_stream);
+                std::vector<std::unique_ptr<cudf::column>> result_cols;
+                result_cols.push_back(
+                    cudf::make_column_from_scalar(
+                        *zero_scalar, 1, chunk_stream, ctx->br()->device_mr()
+                    )
+                );
+                co_await ch_out->send(
+                    rapidsmpf::streaming::to_message(
+                        0,
+                        std::make_unique<rapidsmpf::streaming::TableChunk>(
+                            std::make_unique<cudf::table>(std::move(result_cols)),
+                            chunk_stream
+                        )
+                    )
+                );
+            }
+        }
+        // Non-zero ranks don't send anything (following q09 pattern)
+    } else {
+        // Single rank: divide by 7.0 here
+        if (local_result) {
+            auto sum_val =
+                static_cast<cudf::numeric_scalar<double>*>(
+                    cudf::reduce(
+                        local_result->view().column(0),
+                        *cudf::make_sum_aggregation<cudf::reduce_aggregation>(),
+                        cudf::data_type(cudf::type_id::FLOAT64),
+                        chunk_stream,
+                        ctx->br()->device_mr()
+                    )
+                        .get()
+                )
+                    ->value(chunk_stream);
```

Comment:
this is the same as `local_sum` in a single rank case, isnt it? Seems a bit redundant